caravel/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -40,41 +41,49 @@ class SqlEditorLeftBar extends React.PureComponent {
     if (!(db)) {
       this.setState({ tableOptions: [] });
     } else {
-      this.fetchTables(val, this.props.queryEditor.schema);
       this.fetchSchemas(val);
+      this.fetchTables(val, this.props.queryEditor.schema);
     }
   }
   resetState() {
     this.props.actions.resetState();
   }
-  fetchTables(dbId, schema) {
-    const actualDbId = dbId || this.props.queryEditor.dbId;
-    if (actualDbId) {
-      const actualSchema = schema || this.props.queryEditor.schema;
-      this.setState({ tableLoading: true });
+  getTableNamesBySubStr(input, callback) {
+    this.fetchTables(
+        this.props.queryEditor.dbId,
+        this.props.queryEditor.schema,
+        input);
+    callback(null, { options: this.state.tableOptions });
+  }
+  fetchTables(dbId, schema, substr) {
```

Comment:
yep, agree - I want to do that in a separate PR.
This one is to get caching in place, another one will be  moving fetching data into actions.